### DISCUSSION TITLE
Assistance with Sleep-on-LAN (SOL)

### DISCUSSION THREAD

**Discussion by roomonthethird** [AlexGustafsson/homebridge-wol#253] [Questions]
Hey there! I saw the similar question asked in #216 but I'm still confused how to execute the commands properly.

Currently Homebridge-WOL works only to turn on the computer from sleep. I'd like to be able to set the computer to sleep as well. 
I imagine I have to use something like this; https://github.com/SR-G/sleep-on-lan
But I don't understand how to connect that to the Homebridge-WOL plugin.

I tried writing a simple batch script and executing that through the Shutdown config but I don't think I know what I'm doing lol. 

If anyone could ELI5 that would be really helpful. Thank you!

**Comment by kamui** [comment]
In the shutdown command just put in:

```
curl http://<username>:<password>@<ip>:8009/sleep
```

where username and password are what you configured in Sleep On LAN if any, if not, leave them out and leave out the `:` and the `@` before the `ip`. Put the IP of the computer you installed Sleep On LAN on.

You should be running Sleep On LAN as a service like they recommend with `nssm`. The only thing I had to do in addition was go to Windows Security > Firewall > Allow app through firewall > Change settings > Allow another app > Browse > Then find the sol.exe where you installed it > then check Private if homebridge is on your local network
